_posts/2024-06-25-create-your-first-python-package-scipy-2024.md

```diff
@@ -0,0 +1,168 @@
+2. Run your downloaded file and follow the on-screen instructions.
+3. Restart your terminal.
+4. To verify that the shell can find and run the hatch command in your `PATH`, run `hatch --version` in your Terminal.
+
+#### For Linux users
+*Note: if you prefer to use a tool other than pipx, please refer to the [Hatch documentation](https://hatch.pypa.io/latest/) for more information.*
+* Install hatch from the command line using pipx: `pip install hatch`
+
+#### For Windows users
+*These instructions are for installing Hatch using the GUI installer. If you’d prefer to use the Command line installer, please see the [Hatch documentation](https://hatch.pypa.io/latest/install/#command-line-installer_1).*
+1. In your browser, download the `.msi` file: [hatch-x64.msi](https://github.com/pypa/hatch/releases/latest/download/hatch-x64.msi).
+2. Run your downloaded file and follow the on-screen instructions.
+3. Restart your terminal.
+4. To verify that the shell can find and run the hatch command in your `PATH`, run `hatch --version` in your Terminal.
+
+### Configure Hatch (all systems)
+Once you have installed Hatch, you will want to customize the configuration. Hatch stores your configuration information in a `config.toml` file.
+
+While you can update the `config.toml` file through the command line, it might be easier to look at it and update it in a text editor if you are using it for the first time.
+
+1. Open and edit your `config.toml` file by either:
+  * Running `hatch config explore` in your shell, which will open up a directory window that will allow you to double click on the file and open it in your favorite text editor.
+  * Alternatively, you can retrieve the location of the Hatch config file by running `hatch config find` in your shell.
+2. Update your email and name
+  * Once the file is open, update the \[template\] table of the `config.toml` file with your name and email. This information will be used in any `pyproject.toml` metadata files that you create using Hatch.
+3. Set tests to `false`
+*While tests are important, setting the tests configuration in Hatch to true will create a more complex pyproject.toml file. You won’t need to use this feature in this beginner friendly tutorial series but we will introduce it in later tutorials.*
+  * Set tests to `false` in the `[template.plugins.default]` table.
+4. Close the config file and `run hatch config show`
+`hatch config show` will print out the contents of your config.toml file in your shell. Look at the values and ensure that your name and email are set, and also  make sure that `tests=false`.
+
+## Helpful links
+* Our [example Python package repo, `pyosPackage`](https://github.com/pyOpenSci/pyosPackage), that goes along with pyOpenSci tutorials.
+* [Workshop information on the SciPy 2024 website](https://cfp.scipy.org/2024/talk/QT9GBY/).
+
+## Workshop agenda
```

Comment:
Let's structure this page as follows

1. the what when where at the top like you have!
2. the workshop agenda
3. setup instructions last.

then, can you please see if making the agenda a table looks ok? ideally a table is nice because we can separate out each hour and overview specific events during that hour (and provide links). 

i can style the table to look more nice if it doesn't look nice using our default theme. see if that works / looks more skimmable?

Reply:
What might look nice is to have 

| hour 1 | description here |
| Takeaways of hour 1  |

so in other words a coliumn for time and on the right a column for description. and then between each "section" of the workshop a 2 column spanning row that have the takeaways in a block of text?

you may have to [use html for this to work](https://notes.aliciasykes.com/36402/github-markdown-tricks) i am not sure. the goal is to make the agenda more skimmable and look different from the setup instructions.

Reply:
done. built the tables in HTML.